Disher/View Controllers/ListsViewController.m

```diff
@@ -6,53 +6,76 @@
 //
 
 #import "ListsViewController.h"
+#import "CreateListViewController.h"
 #import "ListCell.h"
+#import "List.h"
 
-@interface ListsViewController () <UITableViewDelegate, UITableViewDataSource>
-
+@interface ListsViewController () <UITableViewDelegate, UITableViewDataSource, ListDelegate>
+@property (weak, nonatomic) IBOutlet UITableView *tableView;
+@property (strong, nonatomic) NSArray *lists;
 @end
 
 @implementation ListsViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
+    self.navigationItem.rightBarButtonItem = self.editButtonItem;
+    self.tableView.delegate = self;
+    self.tableView.dataSource = self;
+    [self queryLists];
     
-    // Uncomment the following line to preserve selection between presentations.
-    // self.clearsSelectionOnViewWillAppear = NO;
-    
-    // Uncomment the following line to display an Edit button in the navigation bar for this view controller.
-    // self.navigationItem.rightBarButtonItem = self.editButtonItem;
 }
 
 #pragma mark - Table view data source
 
-- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
-#warning Incomplete implementation, return the number of sections
-    return 2;
-}
-
 - (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
-#warning Incomplete implementation, return the number of rows
-    return 5;
+    return self.lists.count;
 }
 
-
 - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
     ListCell *cell = [tableView dequeueReusableCellWithIdentifier:@"ListCell" forIndexPath:indexPath];
-    
-    // Configure the cell...
-    
+    List *currentList = self.lists[indexPath.row];
+    cell.listName.text = currentList[@"listName"];
     return cell;
 }
 
-/*
+- (void) queryLists {
+    PFQuery *query = [PFQuery queryWithClassName:@"List"];
+    [query orderByDescending:@"updatedAt"];
+    [query includeKey:@"listName"];
+    [query includeKey:@"recipes"];
+    [query includeKey:@"objectID"];
+    [query findObjectsInBackgroundWithBlock:^(NSArray * _Nullable objects, NSError * _Nullable error) {
```

Comment:
does this API run on the main thread? generally speaking, it's better to not have the `reloadData` directly inside this block, but instead separate it and ensure it runs on main thread.